lib/ansible/module_utils/facts.py

```diff
@@ -3169,8 +3169,44 @@ def populate(self):
         return self.facts
 
     def get_virtual_facts(self):
-        self.facts['virtualization_type'] = ''
-        self.facts['virtualization_role'] = ''
+        sysctl_path = self.module.get_bin_path('sysctl', required=True)
```

Comment:
@bcoca: I tested what would happen if i removed the sysctl binary on a testbox, and noticed the setup-module will already fail, because there are some places where the tool path is hardcoded and this leads to the module erroring out. This means my required=True addition does not make anything worse than it already is.

If the real solution is to also replace other instances of hardcoded sysctl paths (which would have to be a separate PR imo), then my question is what the desired behaviour is when the tool is not found: setting the effected fact variables to the empty string, or not setting them at all.